drivers/samsung/displays/md_series.cr

```diff
@@ -0,0 +1,500 @@
+  # TODO: check type for broadcast is correct
+  def power(power : Bool, broadcast : String? = nil)
+    self[:power_target] = power
+    self[:power_stable] = false
+
+    if !power
+      # Blank the screen before turning off panel if required
+      # required by some video walls where screens are chained
+      switch_to(@blank.as(String)) if @blank && self[:power]?
+      do_send("panel_mute", 1)
+    elsif !@rs232 && !self[:connected]?
+       wake(broadcast)
+    else
+      # Power on
+      do_send("hard_off", 1)
+      do_send("panel_mute", 0)
+    end
+  end
+
+  def hard_off
+    do_send("panel_mute", 0) if self[:power]?
+    do_send("hard_off", 0)
+    do_poll
+  end
+
+  # TODO: figure out what block is for
+  # def power?(**options, &block)
+  def power?(**options)
+    # options[:emit] = block unless block.nil?
+    do_send("panel_mute", [] of UInt8, **options)
+  end
+
+  # Adds mute states compatible with projectors
+  def mute(state : Bool = true)
+    power(!state)
+  end
+
+  def unmute
+    power(true)
+  end
+
+  # check software version
+  def software_version?
+    do_send("software_version")
+  end
+
+  def serial_number?
+    do_send("serial_number")
+  end
+
+  # Converts a hex encoded string into a binary string
+  def byte_to_hex(data : String) : String
+    # Removes invalid characters
+    data = data.gsub(/(0x|[^0-9A-Fa-f])*/, "")
+
+    # Ensure we have an even number of characters
+    data = '0' + data if data.size % 2 > 0
+
+    # Breaks string into an array of characters
+    output = [] of Int32
+    data.scan(/.{2}/) do |md|
+      output << md[0].to_i(16)
+    end
+
+    String.build do |io|
+      output.each do |number|
+        io.write_byte number.to_u8
+      end
+    end
+  end
+
+  # ability to send custom mdc commands via backoffice
+  def custom_mdc(command : String, value : String)
+    command = byte_to_hex(command).bytes[0].to_i
+    data = byte_to_hex(value).bytes
+    do_send(command, data)
+  end
+
+  def set_timer(enable : Bool = true, volume : Int32 = 0)
+    # set the time on the display
+    time_request = [] of Int32
+    t = Time.local
+    time_request << t.day
+    hour = t.hour
+    ampm = if hour > 12
+             hour = hour - 12
+             0 # pm
+           else
+             1 # am
+           end
+    time_request << hour
+    time_request << t.minute
+    time_request << t.month
+    year = t.year.to_s(16).rjust(4, '0')
+    time_request << year[0..1].to_i(16)
+    time_request << year[2..-1].to_i(16)
+    time_request << ampm
+
+    do_send("time", time_request)
+
+    state = enable ? "01" : "00"
+    vol = volume.to_s(16).rjust(2, '0')
+    # on 03:45 am enabled  off 03:30 am enabled  on-everyday  ignore manual  off-everyday  ignore manual  volume 15  input HDMI   holiday apply
+    custom_mdc(
+      "A4",
+      "03-2D-01 #{state}  03-1E-01   #{state}      01          80               01           80          #{vol}        21          01"
+    )
+  end
```

Comment:
This looks to be very specific to certain deploy requirements. For simplicity, drop this from the initial implementation. If / when there's a need to use the devices internal timer (over a trigger), this can be implementation in a more general way that allows defining all device timer actions.

Reply:
removed!